dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/delete-payment/delete-payment.event-handler.test.ts

```diff
@@ -326,4 +327,136 @@ describe('handleUtilisationReportPaymentDeletedEvent', () => {
+    when(mockFindOne).calledWith(PaymentEntity, expect.anything()).mockResolvedValue(paymentEntity);
+    when(mockFindOne).calledWith(FeeRecordEntity, expect.anything()).mockResolvedValue(linkedFeeRecordEntity);
+
+    const mockEventHandler = aMockEventHandler();
+    const feeRecordStateMachine = aMockFeeRecordStateMachine(mockEventHandler);
+    jest.spyOn(FeeRecordStateMachine, 'forFeeRecord').mockReturnValue(feeRecordStateMachine);
+
+    jest.mocked(feeRecordsAndPaymentsMatch).mockResolvedValue(true);
+
+    // Act
+    await handleUtilisationReportDeletePaymentEvent(report, {
+      transactionEntityManager: mockEntityManager,
+      paymentId,
+      requestSource,
+    });
+
+    // Assert
+    expect(mockEventHandler).toHaveBeenCalledTimes(1);
+    expect(mockEventHandler).toHaveBeenCalledWith(
+      expect.objectContaining({
+        payload: expect.objectContaining({
+          hasCorrections: true,
+        }) as FeeRecordPaymentDeletedEvent,
+      }),
+    );
+  });
+
+  it('should set hasCorrections to false when the fee record does not have corrections', async () => {
+    // Arrange
+    const requestSource = aDbRequestSource();
+    const paymentId = 123;
+
+    const report = UtilisationReportEntityMockBuilder.forStatus(RECONCILIATION_IN_PROGRESS).build();
+
+    const linkedFeeRecordEntity = FeeRecordEntityMockBuilder.forReport(report).withId(456).withCorrections([]).build();
+
+    const paymentEntity = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withId(paymentId).withFeeRecords([linkedFeeRecordEntity]).build();
+    when(mockFindOne).calledWith(PaymentEntity, expect.anything()).mockResolvedValue(paymentEntity);
+    when(mockFindOne).calledWith(FeeRecordEntity, expect.anything()).mockResolvedValue(linkedFeeRecordEntity);
+
+    const mockEventHandler = aMockEventHandler();
+    const feeRecordStateMachine = aMockFeeRecordStateMachine(mockEventHandler);
+    jest.spyOn(FeeRecordStateMachine, 'forFeeRecord').mockReturnValue(feeRecordStateMachine);
+
+    jest.mocked(feeRecordsAndPaymentsMatch).mockResolvedValue(true);
+
+    // Act
+    await handleUtilisationReportDeletePaymentEvent(report, {
+      transactionEntityManager: mockEntityManager,
+      paymentId,
+      requestSource,
+    });
+
+    // Assert
+    expect(mockEventHandler).toHaveBeenCalledTimes(1);
+    expect(mockEventHandler).toHaveBeenCalledWith(
+      expect.objectContaining({
+        payload: expect.objectContaining({
+          hasCorrections: false,
+        }) as FeeRecordPaymentDeletedEvent,
+      }),
+    );
+  });
+
+  it('should set hasCorrections for each linked fee record independently', async () => {
+    // Arrange
+    const requestSource = aDbRequestSource();
+    const paymentId = 123;
+
+    const report = UtilisationReportEntityMockBuilder.forStatus(RECONCILIATION_IN_PROGRESS).build();
+
+    const linkedFeeRecordEntityOne = FeeRecordEntityMockBuilder.forReport(report).withId(456).withCorrections([]).build();
```

Comment:
could 456 be a constant as used in 2 tests?

Reply:
But here the tests are independent of each other so if we changed to id in one test we are not at all required to change it in the other tests so I don't see any advantage. There are lots of tests in this file that make use of 456 as some of their data.

Reply:
What is the advantage? The tests are completely idependent, and if we changed the id of one in one test it is not relevant to the other test and would not be necessary to change their either. We're only specifying id in this test because we are creating two fee records and they would otherwise have the same default id and that is not realistic